### PR TITLE
WITH-RSS2 not working with default args

### LINKED ISSUE #6: with-rss2 macro broken
### How to reproduce:

```lisp
(with-output-to-string (s) (with-rss2 (s)))
```

It will fail as:

```lisp
; in: WITH-OUTPUT-TO-STRING (S)
;     (WITH-RSS2 (S))
; --> WITH-XML-OUTPUT LET WITH-TAG LET PROGN XML-EMITTER::START-TAG 
; ==>
;   (("version" "2.0"))
; 
; caught ERROR:
;   illegal function call
; 
; compilation unit finished
;   caught 1 ERROR condition
; Evaluation aborted on #<SB-INT:COMPILED-PROGRAM-ERROR {100926A0D3}>.
```

### Try digging a little deeper:

```lisp
(macroexpand '(with-rss2 (s)))
```

this macro expanded to:

```lisp
(LET ((XML-EMITTER::*XML-OUTPUT-STREAM* S))
  (FORMAT XML-EMITTER::*XML-OUTPUT-STREAM*
          "<?xml version=\"1.0\" encoding=~S~:[~; standalone=\"~:[no~;yes~]\"~]?>~%"
          "ISO-8859-1" NIL NIL)
  (WITH-TAG ("rss" (("version" "2.0")))
    (WITH-TAG ("channel"))))
```

### When do we changed this?

https://github.com/VitoVan/xml-emitter/pull/5/files#diff-5fe6d35e26e9626492ab0952c60beb81R68

It was written as:

```lisp
(with-tag ("rss" '(("version" "2.0"))) ...)
```

now it is expanded as:

```lisp
(WITH-TAG ("rss" (("version" "2.0")))
    (WITH-TAG ("channel")))
```

see the `'(("version" "2.0"))` vs `(("version" "2.0"))`

### How to fix?

1. I tried to add double quotes and it works:

    ```lisp
    (defmacro with-rss2 ((stream &key (encoding "ISO-8859-1") (attrs '(("version" "2.0")))) &body body)
    ```

    to

    ```lisp
    (defmacro with-rss2 ((stream &key (encoding "ISO-8859-1") (attrs ''(("version" "2.0")))) &body body)
    ```

2. Or add quote inside the macro

    ```lisp
    (with-tag ("rss" ,attrs)
           (with-tag ("channel")
             ,@body))
    ``` 

    to

    ```lisp
    (with-tag ("rss" ',attrs)
           (with-tag ("channel")
             ,@body))
    ```

### Question

Should we fix it with extra quotes? Is that ugly? Or am I wrong about this?

Related: https://github.com/VitoVan/xml-emitter/pull/5

And I think it's time to add tests

### PR DESCRIPTION
**Pull request by iamFIREcracker**
## Fix #6 and add unit tests support

Fixing #6 was pretty easy; in fact, double quoting WITH-RSS optional
arguments was all we had to do.  Adding unit tests support to the module
on the other end, wound up being not as easy as I initially thought it
would, but still, I am quite happy with the final result.

I split the tests support in 3 different sub-activities:

1) Add a package (and a system) for the tests: `:xml-emitter/tests`
2) Add a Makefile to run the tests

  This turned out to be more annoying than expected:

  - The interactive debugger needs to be disabled when tests fail, or the
  CI job would hang forever
  - The spawn REPL needs to be shut down when tests pass, or again,
  it would cause the CI job to hang forever
  - Certain implementations did not exit with non-0 code when tests
  failed

  And all this needed to be done outside of the testing system (hence in
  the Makefile) becasue I still wanted to leave things interactive when
  people tried to run the tests with `(asdl:test-system ...)`

3) Add a .travis.yml file to test the library with multiple
   implementations

This is it!

### CODE REVIEW

**Comment by iamFIREcracker** [comment]
@VitoVan - this is as good as I can get it to be.  Give it a go, and let me know if you spot anything weird.

File: tests.lisp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+(defpackage :xml-emitter/tests
+  (:use :common-lisp :xml-emitter))
+
+(in-package :xml-emitter/tests)
+
+(defmacro xml-test (name stream expected &body body)
+  "Utility macro to facilitate testing of the generted xml
```
Comment (VitoVan) [inline tests.lisp]: generted -> generated ?

**Review by VitoVan — DISMISSED** [review]
LGTM

There's 3 more things I planned to do after this is merged:
1. fix typo `generted` -> `generated`
2. change file LICENCE to LICENSE
3. improve travis.yml according to Travis CI config validation info:
   - `root`: missing `os`, using the default `linux`
   - `env`: key `matrix` is an alias for `jobs`, using `jobs`

I hope these are acceptable to you.

And if you don't bother to do it by this PR, I also appreciated that very much.

File: tests.lisp (hunk 1 above)
Comment (iamFIREcracker) [reply tests.lisp]: 3c22e64

**Comment by iamFIREcracker** [comment]
@VitoVan 

3c22e64 Fix typo
829d0fc Fix typo in LICENSE filename
084fb7b Lint .travis.yml

Let me know if you spot anything else.

File: xml-emitter.asd [hunk 2]
```diff
@@ -5,10 +5,10 @@
...
 (in-package #:xml-emitter-system)
 
-(defsystem xml-emitter
+(defsystem :xml-emitter
   :author "Peter Scott / Vito Van / Philipp Marek / Kieran Grant / Matteo Landi"
   :version "1.0.5"
-  :licence "Public Domain / 0-clause MIT"
+  :license "Public Domain / 0-clause MIT"
   :serial t
   :components ((:static-file "LICENCE")
```
Comment (VitoVan) [inline xml-emitter.asd]: `(:static-file "LICENCE")` to `(:static-file "LICENSE")`

Comment (iamFIREcracker) [reply xml-emitter.asd]: 85f2f60

**Comment by iamFIREcracker** [comment]
85f2f60 Fix LICENSE filename inside the .asd file too

**Review by VitoVan — APPROVED** [review]
LGTM

**Comment by VitoVan** [comment]
@iamFIREcracker 

Matteo Landi, You've done a very good job!

I'll bump the version to 1.1.0 later in 24hrs.

**Pull request merged**